core/kazoo_amqp/src/api/kapi_im.erl

```diff
@@ -11,137 +11,210 @@
-                        ,{<<"From">>, fun is_binary/1}
-                        ,{<<"Request">>, fun is_binary/1}
-                        ,{<<"Message-ID">>, fun is_binary/1}
-                        ,{<<"System-ID">>, fun is_binary/1}
-                        ,{<<"Custom-Vars">>, fun kz_json:is_json_object/1}
-                        ,{<<"Body">>, fun is_binary/1}
-                        ,{<<"Endpoints">>, fun is_list/1}
-                        ]).
--define(OUTBOUND_REQ_VALUES, [{<<"Event-Category">>, ?EVENT_CATEGORIES}
-                             ,{<<"Event-Name">>, ?OUTBOUND_REQ_EVENT_NAME}
-                             ,{<<"Route-Type">>, [<<"onnet">>, <<"offnet">>]}
-                             ]).
-
--define(OUTBOUND_RK(Category, RouteId, MsgId), outbound_routing_key(Category, RouteId, MsgId)).
-
--define(BIND_OUTBOUND_RK(Category, Props)
-       ,?OUTBOUND_RK(Category
-                    ,bind_route_id(Props)
-                    ,bind_message_id(Props)
-                    )).
-
--define(PUBLISH_OUTBOUND_RK(Props)
-       ,?OUTBOUND_RK(kz_api:event_category(Props)
-                    ,route_id(Props)
-                    ,message_id(Props)
-                    )).
-
--spec inbound(kz_term:api_terms()) -> {'ok', iolist()} |
-          {'error', string()}.
-inbound(Prop) when is_list(Prop) ->
-    case inbound_v(Prop) of
-        'true' -> kz_api:build_message(Prop, ?INBOUND_HEADERS, ?OPTIONAL_INBOUND_HEADERS);
-        'false' -> {'error', "Proplist failed validation for inbound"}
-    end;
-inbound(JObj) -> inbound(kz_json:to_proplist(JObj)).
+-ifdef(TEST).
+-export([routing_key/4
+        ]).
+-endif.
+
+%%------------------------------------------------------------------------------
+%% @doc Get all API definitions of this module.
+%% @end
+%%------------------------------------------------------------------------------
+-spec api_definitions() -> kapi_definition:apis().
+api_definitions() ->
+    [inbound_definition()
+    ,outbound_definition()
+    ].
+
+%%------------------------------------------------------------------------------
+%% @doc Get API definition of the given `Name'.
+%% @see api_definitions/0
+%% @end
+%%------------------------------------------------------------------------------
+-spec api_definition(kz_term:text()) -> kapi_definition:api().
+api_definition(Name) when not is_binary(Name) ->
+    api_definition(kz_term:to_binary(Name));
+api_definition(<<"inbound">>) ->
+    inbound_definition();
+api_definition(<<"outbound">>) ->
+    outbound_definition().
+
+-spec inbound_definition() -> kapi_definition:api().
+inbound_definition() ->
+    EventName = <<"inbound">>,
+    Category = ?EVENT_CATEGORIES,
+    Setters = [{fun kapi_definition:set_name/2, EventName}
+              ,{fun kapi_definition:set_friendly_name/2, <<"Inbound">>}
+              ,{fun kapi_definition:set_description/2, <<"SMS MMS Inbound">>}
+              ,{fun kapi_definition:set_category/2, Category}
```

Comment:
you're setting a list of binaries in a api_ne_binary

Reply:
Fixed on the new PR.